### PR TITLE
Add some documentation to Scene.construct

### PR DESCRIPTION
**Pull request by leotrs**
This PR adds documentation to some of the classes and methods that are linked to from the first two tutorials. I figured it would be a good idea to start there since they are the most likely to be visited by a newcomer.

## List of Changes
- animation/create.py module docstring
- scene/scene.py::Scene class docstring
- scene/scene.py::Scene.construct method docstring

## Further Comments
The module docstring needs some help. The idea is to have a single quick scene that illustrates all animations within the module. I thought using the logo would be a good idea. Perhaps we should wait until #729 is merged before we move forward with this idea. I'm marking this as draft until then.

cc @ManimCommunity/docs

### CODE REVIEW

**Comment by leotrs** [comment]
Docs build locally but I get the following error on RTD. Halp @ManimCommunity/docs, you're my only hope!

```
ValueError: Your installation does not support converting .pdf files to SVG. Consider updating dvisvgm to at least version 2.4. If this does not solve the problem, please refer to our troubleshooting guide at: https://manimce.readthedocs.io/en/latest/installation/troubleshooting.html

Exception occurred:
  File "/home/docs/checkouts/readthedocs.org/user_builds/manimce/envs/753/lib/python3.8/site-packages/manim/utils/tex_file_writing.py", line 211, in convert_to_svg
    raise ValueError(
ValueError: Your installation does not support converting .pdf files to SVG. Consider updating dvisvgm to at least version 2.4. If this does not solve the problem, please refer to our troubleshooting guide at: https://manimce.readthedocs.io/en/latest/installation/troubleshooting.html
The full traceback has been saved in /tmp/sphinx-err-o56afn9j.log, if you want to report the issue to the developers.
Please also report this if it was a user error, so that a better error message can be provided next time.
A bug report can be filed in the tracker at <https://github.com/sphinx-doc/sphinx/issues>. Thanks!
Command time: 237s Return: 2
```

**Comment by behackl** [comment]
> Docs build locally but I get the following error on RTD. Halp @ManimCommunity/docs, you're my only hope!

Should be fixed with the latest commit in #729.

**Comment by leotrs** [comment]
Depends on #729

Marking as draft until then.

File: manim/animation/creation.py
Comment (kolibril13) [inline manim/animation/creation.py:236]: Maybe we can even add an example for all off these?

Comment (kolibril13) [reply manim/animation/creation.py:236]: And I think this can be used by more Mobjects then only Text.

File: manim/animation/creation.py
Comment (kolibril13) [inline manim/animation/creation.py:310]:
```suggestion
    """Show one submobject at a time, removing all previously displayed ones from screen."""
```
What do you think about including the word "displayed?"

File: manim/scene/scene.py
Comment (kolibril13) [inline manim/scene/scene.py:148]: self.add(t)
                    self.play(Write(t))
I think one of both can be removed

File: manim/scene/scene.py
Comment (kolibril13) [inline manim/scene/scene.py:61]: self.add(t)
                    self.play(Write(t))
I think one of both can be removed

File: manim/animation/creation.py
Comment (leotrs) [reply manim/animation/creation.py:236]: Yes, in the future I would add examples to everything. However, the purpose of this PR is just to have *something* here.

File: manim/scene/scene.py
Comment (leotrs) [reply manim/scene/scene.py:148]: I believe it's a lot better to try to encourage that every mobject is added before it is animated, even if this is not necessary.

File: manim/animation/creation.py
Comment (kolibril13) [reply manim/animation/creation.py:236]: okay, good point!
> And I think this can be used by more Mobjects then only Text. See for example here:
https://docs.manim.community/en/latest/examples.html#threedfunctionplot
self.play(Write(plane))

But then I think It would be nice to add some more text to indicate that Write is not for Text only

Comment (leotrs) [reply manim/animation/creation.py:236]: Oh woops sorry I missed that.

Comment (leotrs) [inline manim/animation/creation.py:236]:
```suggestion
    """Simulate hand-writing a :class:`~.Text` or hand-drawing a :class:`~.VMobject`."""
```

File: manim/animation/creation.py
Comment (behackl) [inline manim/animation/creation.py:51]: Just that I don't get the idea of this wrong: do you intend to remove this once #729 is merged and `ManimBanner` is directly available?

File: manim/animation/creation.py
Comment (behackl) [inline manim/animation/creation.py:186]: "Show" seems somewhat inaccurate, maybe trace or actually draw?

```suggestion
    """Draw the border first and then show the fill."""
```

File: manim/animation/creation.py
Comment (behackl) [inline manim/animation/creation.py:330]:
```suggestion
    """Show a :class:`~.Text` word by word on the scene."""
```

File: manim/animation/creation.py
Comment (behackl) [inline manim/animation/creation.py:284]:
```suggestion
    """Show a :class:`~.Text` letter by letter on the scene.
```

**Review by behackl — COMMENTED** [review]
Only minor remarks; there might be some further ones once the documentation builds successfully.

File: manim/animation/creation.py
Comment (leotrs) [reply manim/animation/creation.py:51]: Yes

File: manim/animation/creation.py
Comment (kolibril13) [inline manim/animation/creation.py:4]: why hide the source code?

Comment (kolibril13) [inline manim/animation/creation.py:4]:
```suggestion
```

File: manim/animation/creation.py
Comment (kolibril13) [inline manim/animation/creation.py:37]: I think this example looks a bit overcrowded and does not show the functionalities in the best way.
https://manimce--753.org.readthedocs.build/en/753/reference/manim.animation.creation.html?highlight=creation#module-manim.animation.creation

File: manim/animation/creation.py
Comment (kolibril13) [inline manim/animation/creation.py:218]:
```suggestion
    """Show one submobject at a time, leaving all previous ones displayed on screen."""
```

File: manim/animation/creation.py
Comment (kolibril13) [inline manim/animation/creation.py:239]: Is this really only for Test? or also for MathTex?

File: manim/scene/scene.py
Comment (kolibril13) [inline manim/scene/scene.py:58]:
```suggestion
                t1 = Text("Hello ")
                t2 = Text("World!").next_to(t1,RIGHT)
                self.add(t1)
                self.play(Write(t2))
```

File: manim/scene/scene.py
Comment (kolibril13) [reply manim/scene/scene.py:145]: I dont think so, as e.g. this script:
```python
class ABC(Scene):
    def construct(self):
        t = Text("Hello World!")
        self.add(t)
        self.wait()
        self.play(Write(t))
```
would not make sense then.
And `self.play(Write(t))` also adds the object to the scene, as you can see in this example
```python
class ABC(Scene):
    def construct(self):
        t = Text("Hello World!")
        self.wait()
        self.play(Write(t))
        self.remove(self.mobjects[-1])
        self.wait()
```

File: manim/scene/scene.py
Comment (kolibril13) [inline manim/scene/scene.py:44]: I don't understand this sentence, which __init__ function should be overwritten?
Can we maybe reference this to an example?

File: manim/animation/creation.py
Comment (leotrs) [reply manim/animation/creation.py:4]: This is shown at the top of the module, so I thought it'd be nice to use this rendered scene as sort of a banner or teaser for the module and its contents. I kinda want to show at a glance what this module is about. The code would be important to show how to recreate this scene, but given that the scene is so niche, I'm not sure that would be helpful to anyone...

File: manim/animation/creation.py
Comment (leotrs) [reply manim/animation/creation.py:37]: I totally agree! This is not finished by any means. I was waiting for ManimBanner to be merged before continuing work here.

File: manim/animation/creation.py
Comment (leotrs) [reply manim/animation/creation.py:239]: I believe it only works for Text, but I might be wrong.

File: manim/scene/scene.py
Comment (leotrs) [reply manim/scene/scene.py:58]: Why? I would rather keep the example simple and to the point.  Since this is for `Scene`, the point is "this is how you create a scene" and not "here is a cool thing you can do with Text"

File: manim/animation/creation.py
Comment (kolibril13) [reply manim/animation/creation.py:4]: Okay, I see, good idea!

File: manim/animation/creation.py
Comment (kolibril13) [reply manim/animation/creation.py:37]: Ahh, perfect!

File: manim/scene/scene.py
Comment (leotrs) [reply manim/scene/scene.py:44]: What I tried to say is that if the user creates a class that inherits from Scene, like `MyScene(Scene)`, they should try not to override the constructor method.

I don't know what kind of example this would link to, as it is mentioning something that _should not_ be done.

File: manim/animation/creation.py
Comment (kolibril13) [reply manim/animation/creation.py:239]: Just tested it, it really *only* works for Text

File: manim/scene/scene.py
Comment (kolibril13) [reply manim/scene/scene.py:44]: Okay, now I get it.
Maybe we can have also a section in the docs which is titled "Best Practice and things to avoid"?

File: manim/scene/scene.py
Comment (kolibril13) [reply manim/scene/scene.py:58]: As mentioned in the next comment, things like
```python
class ABC(Scene):
    def construct(self):
        t = Text("Hello World!")
        self.add(t)
        self.wait()
        self.play(Write(t))
```
could lead to unintended behaviour

File: manim/scene/scene.py
Comment (leotrs) [reply manim/scene/scene.py:44]: That makes sense, I can see a tutorial including that. But that doesn't sound like part of this PR.

File: manim/scene/scene.py
Comment (leotrs) [reply manim/scene/scene.py:58]: I'm not following - what comment? What unintended behavior? :sweat_smile:

Comment (kolibril13) [reply manim/scene/scene.py:58]: See this script:
```python
class ABC(Scene):
    def construct(self):
        t = Text("Hello World!")
        self.wait()
        self.add(t)
        self.wait()
        self.play(Write(t))
```
It once appears, then disappears, then writes.
ABC

Comment (leotrs) [reply manim/scene/scene.py:58]: Oh! Right! Then I'd rather do just `self.play(Write(Text("Hello World!")))`. Is that okay?

Comment (kolibril13) [reply manim/scene/scene.py:58]: sounds good!

**Review by kolibril13 — APPROVED** [review]
I think when https://github.com/ManimCommunity/manim/pull/753#discussion_r533852561
is updated, this one is good to go.

File: manim/scene/scene.py
Comment (leotrs) [reply manim/scene/scene.py]: Done!

**Comment by leotrs** [comment]
The comment mentioned by @kolibril13 has been dealt with!

**Comment by kolibril13** [comment]
perfect!

File: manim/animation/creation.py
Comment (huguesdevimeux) [inline manim/animation/creation.py:129]: **Suggestion**: Specify that it removes the object from the scene as well than showing ShowCreation in reverse.

File: manim/scene/scene.py
Comment (huguesdevimeux) [inline manim/scene/scene.py:130]: **Question:** Are you _sure_ about this? isn't setup stupposed to be used by subclasses of Scene?

**Review by huguesdevimeux — APPROVED** [review]
SGTM !

File: manim/scene/scene.py
Comment (leotrs) [reply manim/scene/scene.py:130]: I believe `setup` is meant to be used by subclasses especifically for the purpose of containing initialization code (that may be shared among different subclasses). But even then, it makes sense for it to also contain the user's initialization code. If a user subclass overrides `setup`, then it's on the user to remember to call `super().setup()` if they need to.

File: manim/animation/creation.py
Comment (leotrs) [reply manim/animation/creation.py:129]: I'm of two minds about this. On the one hand, `Uncreate` has an attribute called `remover` which is set to `True`. If we had better documentation, then that would be enough to document the fact that this Animation removes the mobject.

On the other hand, we of course do _not_ have good documentation and that's not currently documented anywhere. TBH I don't think this is a big deal, and it _will_ get documented at some point, as soon as we document the `Animation.remover` attribute ....

**Comment by leotrs** [comment]
Thanks @kolibril13 for fixing the merge conflicts.

**Pull request merged**
